portfolio/src/main/webapp/script.js

```diff
@@ -12,6 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/*
+ * Fetches message from /data and displays it on the DOM.
+ */
+function getMessage() {
+  fetch('/data').then(response => response.text()).then((mssg) => {
```

Comment:
does response.text() return a promise, or is this just returning the message itself?

Reply:
I believe that it returns the message itself, as text.

Reply:
Then I don't then you need the second then.

fetch('/data').then(response => {
  document.getElementById('message-container').innerText = response.text();
})

Reply:
oh one other thing - I noticed that your screenshot shows the HTML <h1> tags in the response.  You should use innerHTML instead of innerText if the response contains HTML.  Could you follow-up in a new PR?

Reply:
I just tried changing getMessage() to:
`  fetch('/data').then(response => {
    document.getElementById('message-container').innerText = response.text();
  });`

and the output becomes: [object Promise] -- it seems like response.text() is actually a promise, and not text.

Reply:
Sounds good -- I'll open up a new branch/PR for that.